subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -558,13 +732,6 @@ uint8_t ull_adv_sync_pdu_set_clear(struct ll_adv_set *adv,
 		}
 	}
 
-	err = adv_sync_hdr_set_clear(lll_sync, pdu_prev, pdu_new,
-				     hdr_add_fields, hdr_rem_fields,
-				     (data ? data->field_data : NULL));
-	if (err) {
-		return err;
-	}
-
 #if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
 	if (extra_data) {
 		adv_sync_extra_data_set_clear(extra_data_prev, extra_data,
```

Comment:
In my humble opinion if `adv_sync_hdr_set_clear` is extracted from this function, `adv_sync_extra_data_set_clear` should be extracted also. This will make behavior of `ull_adv_sync_pdu_alloc` to be consistent.

`adv_sync_extra_data_set_clear` could be made part of `ull_adv_sync_pdu_set_clear` or it could be extracted to be a global function.

Reply:
for me it looks like extra data has to be handled only once per chain, so doing this inside alloc makes sense. making it part of ull_adv_sync_pdu_set_clear does not make sense because it would be modified on each PDU in chain while making it global means you need to call it every time alloc is done.

Reply:
If you make `adv_sync_hrd_ser_clear` global, then you have to return information about extra_data. Based on that information you decide if call to `adv_sync_hrd_ser_clear` is required or not. Eventual instruction execution overhead makes code readability and maintenance easier. In my humble opinion it is a good tradeoff. 
I suggest to change that to make the code consistent in behaviour. In case there is needed such optimisation on single instructions level, then we may change that current version.